client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -465,7 +465,8 @@ static Request reindex(ReindexRequest reindexRequest) throws IOException {
         Params params = new Params(request)
             .withRefresh(reindexRequest.isRefresh())
             .withTimeout(reindexRequest.getTimeout())
-            .withWaitForActiveShards(reindexRequest.getWaitForActiveShards());
+            .withWaitForActiveShards(reindexRequest.getWaitForActiveShards())
+            .withRequestsPerSecond(reindexRequest.getRequestsPerSecond());
```

Comment:
hmmmm - I think maybe this should be fixed on update by query and delete by query too.

Reply:
I opened a separate PR for this change (https://github.com/elastic/elasticsearch/pull/33808), can you review that one first and we can add this change there already? Its a bit unrelated to adding the rethrottle API so I factored it out.

Reply:
I added this in a commit there.